launch_xml/README.md

```diff
@@ -0,0 +1,92 @@
+# launch_xml
+
+This package provides an abstraction of the XML tree.
+
+## XML front-end mapping rules
+
+### Accessing xml attributes
+
+When having an xml tag like:
+
+```xml
+<tag attr='2'/>
+```
+
+If the entity `e` is wrapping it, the following two statements would be true:
+```python
+hasattr(e, 'attr') == True
+e.attr == '2'
+```
+
+As a general rule, the value of the attribute is returned as an string.
+Conversion to `float` or `int` should be explicitly done in the parser method, or in a substitution.
+For handling lists, see `Built-in Substitutions` section.
+
+### Accessing XML children that aren't an action or a substitution:
+
+In this xml:
+
+```xml
+<executable cmd="ls">
+    <env name="a" value="100"/>
+    <env name="b" value="stuff"/>
+</node>
+```
+
+The `env` childs could be accessed like:
+
+```python
+len(e.env) == 2
+e.env[0].name == 'a'
+e.env[0].value == '100'
+e.env[1].name == 'b'
+e.env[1].value == 'stuff'
+```
+
+In these cases, `e.env` is a list of entity, which could be accessed in the same abstract way.
+
+### Accessing XML children that are an action or a substitution:
+
+The entity will keep track of which `children` where accessed or not.
+The ones which have never been accessed, will be returned when doing:
+
+```python
+e.children
+```
+
+This call should be done after all the children that shouldn't be parsed have been accessed.
+
+### Built-in substitutions
+
+`$(list [sep=,] a,b,c,d)`
```

Comment:
@ivanpauno looking at this yet another time, (1) since spaces are the usual delimiters for substitution arguments, the same should apply here, (2) substitutions currently do not (but may) support keyword arguments and (3) I think this won't do when an action just needs list of things at construction time instead of a substitution which can't be resolved till much later at execution time.